SpatialGDK/Source/SpatialGDK/Private/Interop/SkeletonEntities.cpp

```diff
@@ -0,0 +1,501 @@
+
+	TArray<ComponentData> SkeletonEntityComponentDatas;
+	Algo::Transform(EntityComponents, SkeletonEntityComponentDatas, Convert);
+
+	ViewCoordinator& Coordinator = NetDriver->Connection->GetCoordinator();
+	const Worker_RequestId CreateEntityRequestId =
+		Coordinator.SendCreateEntityRequest(MoveTemp(SkeletonEntityComponentDatas), ActorEntityId);
+
+	CreateEntityDelegate OnCreated = CreateEntityDelegate::CreateLambda(
+		[this, ActorEntityId, WeakActor = MakeWeakObjectPtr(&Actor)](const Worker_CreateEntityResponseOp&) {
+			RemainingSkeletonEntities.Remove(ActorEntityId);
+			if (WeakActor.IsValid())
+			{
+				SkeletonEntitiesToDelegate.Emplace(MakeTuple(ActorEntityId, WeakActor));
+				UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("EntityId: %lld Created skeleton entity"), ActorEntityId);
+			}
+			if (RemainingSkeletonEntities.Num() == 0)
+			{
+				Stage = EStage::DelegatingEntities;
+			}
+		});
+	CreateHandler.AddRequest(CreateEntityRequestId, MoveTemp(OnCreated));
+
+	return ActorEntityId;
+}
+
+void FDistributedStartupActorSkeletonEntityCreator::Advance()
+{
+	CreateHandler.ProcessOps(NetDriver->Connection->GetWorkerMessages());
+
+	if (Stage == EStage::DelegatingEntities)
+	{
+		for (auto Kvp : SkeletonEntitiesToDelegate)
+		{
+			if (!Kvp.Value.IsValid())
+			{
+				continue;
+			}
+
+			const Worker_EntityId CreatedSkeletonEntityId = Kvp.Key;
+			AActor& SkeletonEntityActor = *Kvp.Value;
+
+			const VirtualWorkerId AuthWorkerId = NetDriver->LoadBalanceStrategy->WhoShouldHaveAuthority(SkeletonEntityActor);
+
+			const Worker_PartitionId AuthWorkerPartition =
+				NetDriver->VirtualWorkerTranslator->GetPartitionEntityForVirtualWorker(AuthWorkerId);
+
+			UE_LOG(LogSpatialSkeletonEntityCreator, Log,
+				   TEXT("EntityId: %lld Delegating skeleton entity to VirtualWorker %d Partition %lld"), CreatedSkeletonEntityId,
+				   AuthWorkerId, AuthWorkerPartition);
+
+			AuthorityDelegation AuthorityDelegationComponent;
+			AuthorityDelegationComponent.Delegations.Emplace(SpatialConstants::SERVER_AUTH_COMPONENT_SET_ID, AuthWorkerPartition);
+
+			EntitiesToPopulatingWorkers.Emplace(CreatedSkeletonEntityId, AuthWorkerId);
+
+			NetDriver->Connection->GetCoordinator().SendComponentUpdate(
+				CreatedSkeletonEntityId,
+				ComponentUpdate(OwningComponentUpdatePtr(AuthorityDelegationComponent.CreateAuthorityDelegationUpdate().schema_type),
+								AuthorityDelegation::ComponentId),
+				{});
+		}
+
+		SkeletonEntitiesToDelegate.Empty();
+
+		UE_LOG(LogSpatialSkeletonEntityCreator, Log, TEXT("Finished delegating entities, moving on to manifest"));
+
+		Stage = EStage::SigningManifest;
+	}
+
+	if (Stage == EStage::SigningManifest)
+	{
+		TMap<VirtualWorkerId, TArray<Worker_EntityId>> WorkerToEntitiesToPopulate;
+		// Every worker should get a manifest to proceed through the skeleton entity flow.
+		for (const auto& VirtualWorker : NetDriver->WellKnownEntitySystem->VirtualWorkerTranslationManager->GetVirtualWorkerMapping())
```

Comment:
This is not needed when working with virtual worker Ids since they are integers from 1 to N

Reply:
I thought it'd be more concise this way - iterating on a collection of existing virtual workers, I mean; I'm not really sure if I can rely on 1..N necessarily, or if we're going to change this soon-ish.